test/e2e/node_drain.go

```diff
@@ -641,3 +699,26 @@ func getDrainAndDeleteInterval(deleteInterval []interface{}, drainTimeout *metav
 	res := []interface{}{intervalDuration.String(), deleteInterval[1]}
 	return res
 }
+
+func unblockNodeVolumeDetachment(ctx context.Context, bootstrapClusterProxy framework.ClusterProxy, cluster *clusterv1.Cluster, waitControlPlaneIntervals, waitWorkerNodeIntervals []interface{}) {
+	By("Set NodeVolumeDetachTimeout to 1s to unblock waiting for volume detachments")
+	// Note: This also verifies that KCP & MachineDeployments are still propagating changes to NodeVolumeDetachTimeout down to
+	// Machines that already have a deletionTimestamp.
```

Comment:
Might be we should do a mix.
For CP machines we set NodeVolumeDetachTimeout to 1, for workers we delete the fake volume

Reply:
We can't (at least for CAPD) because we don't have a controller which really does so.